src/test/java/service/DataProcessorFruitsImplTest.java

```diff
@@ -0,0 +1,37 @@
+package service;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Collections;
+import java.util.Map;
+import models.Fruit;
+import org.junit.Test;
+
+public class DataProcessorFruitsImplTest {
+    private static DataProcessor<Map<Fruit, Integer>, String> processor
+            = new DataProcessorFruitsImpl();
+    private static Map<Fruit, Integer> mapOfFruits = Map.of(
+            new Fruit("banana"), 100,
+            new Fruit("tangerine"), 200,
+            new Fruit("apple"), 400);
+
+    @Test
+    public void processWithMap_Ok() {
+        String result = processor.process(mapOfFruits);
+        StringBuilder builder = new StringBuilder();
+        for (Map.Entry<Fruit, Integer> entry : mapOfFruits.entrySet()) {
+            builder.append(entry.getKey().getName())
+                    .append(",")
+                    .append(entry.getValue())
+                    .append(System.lineSeparator());
+        }
```

Comment:
you are duplicating the method logic - let's append actual strings to stringBuilder